### ISSUE TITLE
modules are inferred as distinct types 

### ISSUE THREAD

**Issue by tekumara** [microsoft/pyright#1083]
```python
import os
import sys

modules = [ os, sys ]
```

If `"strictListInference": false` then `modules` is inferred as `List[Module | Module]`

I was expecting `modules` to be inferred as `List[Module]` or `List[ModuleType]`.

This becomes a problem when `"strictListInference": false` because `modules` is inferred as `List[Unknown]` so the types are effectively not checked.

pright 1.1.76

**Comment by erictraut** [comment]
I think the current behavior is correct from an inference standpoint. From a type-checking perspective, different modules have very little in common, so a list of modules should result in `List[Unknown]` if `strictListInference` is false. If `strictListInference` is true, then it should result in a union of the individual modules.

I agree that the string `List[Module | Module]` is confusing. I've changed the code to include the module name, so it will now result in the string `List[Module("os"), Module("sys")]`.

If you want `modules` to be a `List[ModuleType]` you can provide a type declaration:
```python
modules: List[ModuleType] = [os, sys]
```

**Comment by erictraut** [comment]
I've addressed the confusing type string. This is now addressed in Pyright 1.1.77, which I just published. The other behavior is as intended.
